app/navigation/CharityBeneficiaryNavigator.scala

```diff
@@ -86,4 +96,17 @@ class CharityBeneficiaryNavigator extends Navigator {
     }
   }
 
+  private def navigateToAnswersOrAddressQuestions(draftId: String, index: Int): PartialFunction[ReadableUserAnswers, Call] = {
+    case ua =>
+      if (is5mldNonTaxable(ua)) {
+        CharityAnswersController.onPageLoad(index, draftId)
+      } else {
+        AddressYesNoController.onPageLoad(index, draftId)
+      }
+  }
+
+  private def is5mldNonTaxable(ua: ReadableUserAnswers): Boolean = {
+    ua.is5mldEnabled && !ua.isTaxable
+  }
```

Comment:
Could you move this to `Navigator`? Then the duplicate method can be removed from this navigator and the TrustBeneficiaryNavigator

Reply:
Updated